### PR TITLE
Update appui learning

### PR DESCRIPTION
**Pull request by NancyMcCallB**
First installment reviewing AppUI documentation and updating it.

### CODE REVIEW

File: docs/learning/ui/AugmentingUI.md [hunk 1]
```diff
@@ -32,72 +32,44 @@ A recommended technique is for a package to have an initialize() method that reg
...
+Starting in version 2.17 Widgets can support being "popped-out" to a child window by setting the Widget property `canPopout` to true. This option must be explicitly set because the method `getWidgetContent` must return React components that works properly in a child window. At minimum  components should typically not use the `window` or `document` property to register listeners as these listener will be registered for events in the main window and not in the child window. Components will need to use the `ownerDocument` and `ownerDocument.defaultView` properties to retrieve `document` and `window` properties for the child window.
```
Comment (aruniverse) [inline docs/learning/ui/AugmentingUI.md:66]: idk if we want to mention 2.17 here

File: docs/learning/ui/AugmentingUI.md [hunk 2]
```diff
@@ -121,7 +93,7 @@ One last thing to point out in the above example. We specified default size for
 
 ## UiItemProviderOverrides
 
-When registering a [UiItemsProvider]($appui-abstract) with the [UiItemsManager]($appui-abstract) it is possible to pass an additional argument to limit when the provider is allowed to provide its items. The interface [UiItemProviderOverrides]($appui-abstract) defines the parameters that can be used to limit when the provider is called to provide its items.
+When registering a [UiItemsProvider]($appui-react) with the [UiItemsManager]($appui-react) it is possible to pass an additional argument to limit when the provider is allowed to provide its items. The interface [UiItemProviderOverrides]($appui-react) defines the parameters that can be used to limit when the provider is called to provide its items.
```
Comment (aruniverse) [inline docs/learning/ui/AugmentingUI.md:96]: i didnt realize UiItemsManager static didnt move into a property of UiFramework.
not much we can do now, just mentioning

**Review by raplemie — APPROVED** [review]
LGTM!

**Comment by NancyMcCallB** [comment]
https://github.com/Mergifyio backport release/4.0.x

**Pull request merged**